### PR TITLE
Exercise 2

### PR DESCRIPTION
**Pull request by daoduong-saritasa**

### CODE REVIEW

**Review by pavelzadorin-saritasa — CHANGES_REQUESTED** [review]
Your API falls with an error on incorrect user data.

And by architecture, you are using a direct database access from the endpoint controller, those. in other words, you work with repository directly from view, this is bad-practice: https://www.oreilly.com/library/view/software-architecture-patterns/9781491971437/ch01.html. Please look again at the approach with active record pattern.

File: ex2/api/models/user.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import client from "../config/db.config.js";
+
+export class User {
+  #email;
```
Comment (pavelzadorin-saritasa) [inline ex2/api/models/user.js:4]: I've never seen this syntax using, this is it: http://devdoc.net/web/developer.mozilla.org/en-US/docs/JavaScript/Sharp_variables_in_JavaScript.html? but then this feature is obsolete and not recomended to use.
But you can use actual class field feature: https://javascript.info/class#class-fields

File: ex2/api/models/user.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  async save() {
+    const queryString = this.#getSaveQuery();
+    await client.query(queryString);
+  }
+
+  static async findFirst(field, value) {
+    const queryString = this.#getFindQuery(field, value);
+    const result = await client.query(queryString);
+    return result.rows[0];
```
Comment (pavelzadorin-saritasa) [inline ex2/api/models/user.js:35]: I think just returning a result from the database is not safe, with changes to the structure on the database side, your application will receive incorrect data and strange errors.
I am sure that the approach using model mappers (as we do in front-end solutions when processing data from the API) would be perfect in this case too.

**Review by pavelzadorin-saritasa — APPROVED** [review]

File: ex2/api/models/user.js (hunk 1 above)
Comment (daoduong-saritasa) [reply ex2/api/models/user.js:4]: This syntax is [private declaration](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields) in javascript class

Comment (pavelzadorin-saritasa) [reply ex2/api/models/user.js:4]: Ou, ok, the feature was included in the standard only last year), but it seems that all browsers already support it.

**Pull request merged**